src/front/app/routes/register.js

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+import propTypes from 'prop-types';
+import { TextField, PrimaryButton, MessageBar, MessageBarType } from 'office-ui-fabric-react';
+import apiCall from '../apiCall';
+
+class RegisterRoute extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.handleSubmit = this.handleSubmit.bind(this);
+        this.handleChange = this.handleChange.bind(this);
+
+        this.state = {
+            email: '',
+            password: '',
+            error_email: false,
+            error_password: false,
+            success: false,
+        };
+    }
+
+    handleSubmit(event) {
+        event.preventDefault();
+        this.setState(prevState => ({
+            ...prevState,
+            error_email: false,
+            error_password: false,
+            success: false,
+            isSubmitting: true,
+        }));
+        apiCall('post', 'auth/register', {
+            email: this.state.email,
+            password: this.state.password,
+        })
+            .then(() => {
+                this.setState(prevState => ({
+                    ...prevState,
+                    success: true,
+                }));
+                setTimeout(() => {
+                    this.props.history.push('/login');
+                }, 2000);
+            })
+            .catch(err => {
+                if (err.response) {
+                    if (err.response.data === 'validation/email') {
```

Comment:
Very nice, actually handling the different responses s.t. the client has good UI.